service/storage_service.cc

```diff
@@ -6842,28 +6842,59 @@ void storage_service::init_messaging_service() {
             return ss.raft_topology_cmd_handler(term, cmd_index, cmd);
         });
     });
-    ser::storage_service_rpc_verbs::register_raft_pull_snapshot(&_messaging.local(), [handle_raft_rpc] (raft::server_id dst_id, raft_snapshot_pull_params params) {
-        return handle_raft_rpc(dst_id, [params = std::move(params)] (storage_service& ss) -> future<raft_snapshot> {
+    ser::storage_service_rpc_verbs::register_raft_pull_snapshot(&_messaging.local(), [handle_raft_rpc, &feat = _feature_service] (raft::server_id dst_id, raft_snapshot_pull_params params) {
+        return handle_raft_rpc(dst_id, [params = std::move(params), &feat] (storage_service& ss) -> future<raft_snapshot> {
             utils::chunked_vector<canonical_mutation> mutations;
             // FIXME: make it an rwlock, here we only need to lock for reads,
             // might be useful if multiple nodes are trying to pull concurrently.
             auto read_apply_mutex_holder = co_await ss._group0->client().hold_read_apply_mutex(ss._abort_source);
 
-            // We may need to send additional raft-based tables to the requester that
-            // are not indicated in the parameter.
-            // For example, when a node joins, it requests a snapshot before it knows
-            // which features are enabled, so it doesn't know yet if these tables exist
-            // on other nodes.
-            // In the current "legacy" mode we assume the requesting node sends 2 RPCs - one for
-            // topology tables and one for auth tables, service levels, and additional tables.
-            // When we detect it's the second RPC, we add additional tables based on our feature flags.
-            // In the future we want to deprecate this parameter, so this condition should
-            // apply only for "legacy" snapshot pull RPCs.
+            // The RPC's semantics changed at some point and now we support two "modes":
+            //
+            // - Legacy: RPC sender specifies which tables to fetch in the params.tables parameter.
```

Comment:
Shouldn't "legacy" assume that the sender doesn't know that there are two modes at all? I.e., we should support a sender that is using older version of Scylla (mixed cluster) so they will in particular send only one RPC call